keps/sig-node/20200203-memory-manager.md

```diff
@@ -446,38 +478,70 @@ InternalContainerLifecycle will call out Memory Manager on container lifecycle e
 
 ### Feature Gate and Kubelet Flags
 
-A new feature gate will be added to enable the Memory Manager feature. This feature gate will be disabled by default in the Alpha release.
+A new feature gate will be added to enable the Memory Manager feature. This feature gate will be disabled by default in the early releases.
 
-- Proposed Feature Gate:  
-  `--feature-gate=MemoryManager=true`
+#### Feature Gate  
+`--feature-gate=MemoryManager=true`
+
+#### Memory Manager Policy Flag  
+`--memory-manager-policy` - to configure Memory Manager behaviour, `static` enables cross-NUMA feature, whereas `none` disables it 
+
+Example:
+`--memory-manager-policy=none` or `--memory-manager-policy=static`
+
+#### Pre-Reserved Memory Zone Flag
 
 This will be also followed by a Kubelet Flag for the Memory Manager configuration, which is described above in [The configuration of Memory Manager](#the-configuration-of-memory-manager) section.
-- Proposed Configuration Flag:  
-  `--pre-reserved-zone=[{numa-node=int, memory-type=string, limit=string}][,][...]`
-- The example of configuration flag:
-   `--pre-reserved-zone= {numa-node=0, memory-type=memory, limit=1Gi}, {numa-node=1, memory-type=memory, limit=1Gi}`
+
+`--pre-reserved-memory-zone=[{numa-node=int, memory-type=string, limit=string}][,][...]`
+
+Example:
+   `--pre-reserved-memory-zone= {numa-node=0, memory-type=memory, limit=1Gi}, {numa-node=1, memory-type=memory, limit=1Gi}`
+
+#### Cross-NUMA Groups Flag
```

Comment:
> Correct me if I'm wrong. One "cross-NUMA group" it's one cpuset cgroup instance, right? You introduced "cross-NUMA group" since you faced with overlap problem, right?

We introduced this flag, just for the purpose to give way to a user to configure preferred nodes for multiple NUMA node memory allocation. This is an optional flag, and if it not provided we will just follow the minimal number of NUMA nodes pattern.

> Did you asked in linux kernel mailing list about extending this cgroup controller. Of course nobody knows when it will happen even linux kernel community will be agree with that change, but technically it's possible.

Can you please clarify what do you mean by extending the cgroup controller?



> Also is cpuset.mem_exclusive planned here or not?

Currently no. We want to have the same behavior as we have under the CPU manager.

Reply:
> > Correct me if I'm wrong. One "cross-NUMA group" it's one cpuset cgroup instance, right? You introduced "cross-NUMA group" since you faced with overlap problem, right?
> 
> We introduced this flag, just for the purpose to give way to a user to configure preferred nodes for multiple NUMA node memory allocation. This is an optional flag, and if it not provided we will just follow the minimal number of NUMA nodes pattern.
The question was not about the option, it was about additional entity cross-NUMA.
> 
> > Did you asked in linux kernel mailing list about extending this cgroup controller. Of course nobody knows when it will happen even linux kernel community will be agree with that change, but technically it's possible.
> 
> Can you please clarify what do you mean by extending the cgroup controller?
Did you ask in lkml about their future plan on cpuset cgroup controller or another controller in cgroup v2?
> 
> > Also is cpuset.mem_exclusive planned here or not?
> 
> Currently no. We want to have the same behavior as we have under the CPU manager.
ok, but cpuset.cpu_exclusive it's about core, but cpuset.mem_exclusive about a NUMA nodes.

Reply:
> The question was not about the option, it was about additional entity cross-NUMA.

Sorry, I just was unsure what did you mean by `one cpuset cgroup instance`, the cross NUMA groups will create an abstract memory pinning group, but it will not create a new hierarchy under cgroups.

> Did you ask in lkml about their future plan on cpuset cgroup controller or another controller in cgroup v2?

As far as I know, we did not ask, maybe @bg-chun has more information.

> Currently no. We want to have the same behavior as we have under the CPU manager.

I meant that the CPU manager does not use exclusive, so, for now, we will also avoid to use it. I believe using the `cpuset.mem_exclusive` can be considered as additional feature under the memory manager.
.